### PR TITLE
Add an abstract Path class for both static asset and SD card pure pursuit

### PR DESCRIPTION
**Pull request by Exortions**
#### Summary
Add an abstract Path class for both static asset and SD card pure pursuit.
There is now a `Path` class that can be directly passed to `chassis.follow`.
The class has two implementations: `AssetPath` and `FilePath`. This allows pure pursuit to read from both static asset and the SD card.

#### Motivation
It was discussed in the server, as being able to have the flexibility of either SD card or static asset is useful.

#### Test Plan
This still needs to be tested, I don't have access to a chassis to test this with.

## Download the template for this pull request: 

> [!NOTE]  
> This is auto generated from [`Add Template to Pull Request`](https://github.com/LemLib/LemLib/actions/runs/6973082424)
- via manual download: [LemLib@0.5.0-rc1-c8d1f4e.zip](https://nightly.link/LemLib/LemLib/actions/artifacts/1070001070.zip)
- via PROS Integrated Terminal: 
 ```
curl -o LemLib@0.5.0-rc1-c8d1f4e.zip https://nightly.link/LemLib/LemLib/actions/artifacts/1070001070.zip;
pros c fetch LemLib@0.5.0-rc1-c8d1f4e.zip;
pros c apply LemLib@0.5.0-rc1-c8d1f4e;
rm LemLib@0.5.0-rc1-c8d1f4e.zip;
```

### CODE REVIEW

File: include/lemlib/movements/purepursuit.hpp [hunk 1]
```diff
@@ -23,7 +25,7 @@ class PurePursuit : public Movement {
          * @param reversed whether the chassis should move forwards (false) or backwards (true)
          * @param maxSpeed the maximum speed the robot can move at. 127 by default
          */
-        PurePursuit(float trackWidth, const asset& path, float lookaheadDist, int timeout, bool reversed, int maxSpeed);
+        PurePursuit(float trackWidth, Path path, float lookaheadDist, int timeout, bool reversed, int maxSpeed);
```
Comment (liam-teale) [inline include/lemlib/movements/purepursuit.hpp]:
```suggestion
        PurePursuit(float trackWidth, Path& path, float lookaheadDist, int timeout, bool reversed, int maxSpeed);
```
Copy the path by reference

**Review by liam-teale — CHANGES_REQUESTED** [review]
Move semantics need changes

**Comment by liam-teale** [comment]
I'm not sure about the method by which this PR implements the functionality. While I do agree that the feature that this PR implements is required, I think that simply passing a vector of waypoints to pure pursuit will be better, and having functions which return a vector of waypoints given the asset name or file path.

**Comment by Exortions** [comment]
I'll get it done tonight

**Comment by liam-teale** [comment]
is this still active?

**Pull request closed without merging**